src/components/dialog/dialog.spec.ts

```diff
@@ -0,0 +1,346 @@
+import {
+  elementUpdated,
+  expect,
+  fixture,
+  html,
+  unsafeStatic,
+} from '@open-wc/testing';
+import sinon from 'sinon';
+import { defineComponents, IgcDialogComponent } from '../../index.js';
+
+describe('Dialog component', () => {
+  const fireMouseEvent = (type: string, opts: MouseEventInit) =>
+    new MouseEvent(type, opts);
+  const getBoundingRect = (el: Element) => el.getBoundingClientRect();
+
+  before(() => {
+    defineComponents(IgcDialogComponent);
+  });
+
+  let dialog: IgcDialogComponent;
+  let dialogEl: HTMLDialogElement;
+
+  describe('', () => {
+    beforeEach(async () => {
+      dialog = await createDialogComponent();
+      dialogEl = dialog.shadowRoot!.querySelector(
+        'dialog'
+      ) as HTMLDialogElement;
+    });
+
+    it('passes the a11y audit', async () => {
+      const dialog = await fixture<IgcDialogComponent>(
+        html`<igc-dialog></igc-dialog>`
+      );
+
+      await expect(dialog).shadowDom.to.be.accessible();
+    });
+
+    it('should render content inside the dialog', async () => {
+      const content = 'Dialog content';
+      dialog = await createDialogComponent(
+        `<igc-dialog><span>${content}</span></igc-dialog>`
+      );
+
+      dialog.show();
+      await elementUpdated(dialog);
+
+      expect(dialog).dom.to.have.text(content);
+      expect(dialog).dom.to.equal(
+        `
+        <igc-dialog>
+          <span>
+            Dialog content
+          </span>
+        </igc-dialog>`,
```

Comment:
Confused. No way this is actually testing if the dialog has the content since it's not looking in the shadow DOM - so not really checking if the content is inside the dialog for real.
Same line of thinking for the second check - which again tests the light DOM for the very same nodes that were used in `createDialogComponent`, thus testing if the fixture and native DOM node creation works, kinda pointless.

Reply:
PS: Tested by deleting the default slot in the dialog, this test still passed :)